### PR TITLE
update MsgUpgradeTimeout response, add getter directives

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

part of the work needed for #3855 

similar changes to [this PR to main](https://github.com/cosmos/ibc-go/pull/3856/files#diff-1ee551ae5fd35d668a0d5611c204233cb0f73b98f57421fc888cbce1675a9b19) but for the `Upgrade` messages.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/channel/v1/tx.proto [hunk 1]
```diff
@@ -275,6 +275,9 @@ message MsgChannelUpgradeInit {
 
 // MsgChannelUpgradeInitResponse defines the MsgChannelUpgradeInit response type
 message MsgChannelUpgradeInitResponse {
+  option (gogoproto.equal)           = false;
```
Comment (DimitrisJim) [inline proto/ibc/core/channel/v1/tx.proto:278]: weird how this didn't result in any [`Equal`](https://pkg.go.dev/github.com/gogo/protobuf/plugin/equal) being removed from the diff. Are we sure they aren't disabled by default? I wonder why we initially started using it in other places.

**Review by DimitrisJim — APPROVED** [review]
lgtm granted #3856 is approved

File: proto/ibc/core/channel/v1/tx.proto (hunk 1 above)
Comment (charleenfei) [reply proto/ibc/core/channel/v1/tx.proto:278]: good point! i can't find it in the file at all. i've seen references to it in the SDK but always for setting `Equal` to true, i think for `Coin` messages. and looking into the docs about this directive, [equal does seem to be disabled by default.](https://pkg.go.dev/github.com/gogo/protobuf/plugin/equal)

Comment (colin-axner) [reply proto/ibc/core/channel/v1/tx.proto:278]: I was wondering the same. I'd be in favour of removing the equal directives set to false

Comment (crodriguezvega) [reply proto/ibc/core/channel/v1/tx.proto:278]: When I added this option in `MsgRegisterInterchainAccountResponse` in [this other PR](https://github.com/cosmos/ibc-go/pull/3856), I see that the getter functions get removed [here](https://github.com/cosmos/ibc-go/pull/3856/files#diff-9f4f1d3ff2840c8fe4c2a21f8907b11aad02a4c43d84d2855b0fdafe6fd7d1b9)...

**Pull request merged**